app/src/ui/diff/index.tsx

```diff
@@ -1044,52 +1124,29 @@ export class Diff extends React.Component<IDiffProps, {}> {
     this.codeMirror = cmh === null ? null : cmh.getEditor()
   }
 
-  public render() {
-    const diff = this.props.diff
-
-    if (diff.kind === DiffType.Image) {
-      return this.renderImage(diff)
-    }
-
-    if (diff.kind === DiffType.Binary) {
-      return this.renderBinaryFile()
-    }
-
-    if (diff.kind === DiffType.TooLarge) {
-      const BlankSlateImage = encodePathAsUrl(
-        __dirname,
-        'static/empty-no-file-selected.svg'
-      )
-      const diffSizeMB = Math.round(diff.length / (1024 * 1024))
-      return (
-        <div className="panel empty">
-          <img src={BlankSlateImage} className="blankslate-image" />
-          The diff returned by Git is {diffSizeMB}MB ({diff.length} bytes),
-          which is larger than what can be displayed in GitHub Desktop.
-        </div>
-      )
-    }
-
-    if (diff.kind === DiffType.Text) {
-      if (diff.hunks.length === 0) {
-        if (this.props.file.status === AppFileStatus.New) {
-          return <div className="panel empty">The file is empty</div>
-        }
-
-        if (this.props.file.status === AppFileStatus.Renamed) {
-          return (
-            <div className="panel renamed">
-              The file was renamed but not changed
-            </div>
-          )
-        }
-
-        return <div className="panel empty">No content changes found</div>
+  private renderDiff(diff: IDiff): JSX.Element | null {
+    switch (diff.kind) {
+      case DiffType.Text:
+        return this.renderText(diff)
+      case DiffType.Binary:
+        return this.renderBinaryFile()
+      case DiffType.Image:
+        return this.renderImage(diff)
+      case DiffType.LargeText: {
+        const largeTextDiff = diff as ILargeTextDiff
```

Comment:
this is actually an open question:
The above methods that take `diff`, ex: `this.renderText(diff)` don't need to cast diff as `ITextDiff` because the method `renderText` checks that diff is `ITextDiff` in its input. Accordingly, is there a special need to cast diff as `ILargeTextDiff`? it looks like the `renderLargeText` method expects that type as its input anyway.

If we already talked about this, I'm sorry! This looks vaguely familiar.

Reply:
This was a leftover from refactoring `ILargeTextDiff` to not have multiple purposes. Now that we have `IUnrenderableDiff` we don't have to worry about casting any longer.